src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/ServiceController.php

```diff
@@ -318,8 +319,18 @@ public function adminOverviewAction($serviceId)
         $service = $this->authorizationService->changeActiveService($serviceId);
         $entityList = $this->entityService->getEntityListForService($service);
         $serviceList = new ServiceList([Service::fromService($service, $entityList, $this->router)]);
-        $productionEntitiesEnabled = [$service->isProductionEntitiesEnabled()];
+        $productionEntityEnabled = $service->isProductionEntitiesEnabled();
         $privacyOK = [$this->serviceStatusService->hasPrivacyQuestions($service)];
+        $hasTestEntities = false;
+
+        foreach ($serviceList as $service) {
+            if ($service->hasTestEntities()) {
+                $hasTestEntities = true;
+                break;
+            }
+        }
+
+        $productionEntitiesEnabled = [$productionEntityEnabled || $hasTestEntities];
```

Comment:
Again, maybe key this with the serviceId on the array. And while looking at this change. You should be able to move all the logic you added in the controllers to the service itself. Cleaning up the controller logic significantly.

Reply:
no longer need this array after moving the logic to the viewobject entirely 👍